internal/notifier/bitbucketserver.go

```diff
@@ -268,6 +268,15 @@ func parseBitbucketServerGitAddress(s string) (string, string, error) {
 	}
 	//Remove "scm/" --> https://community.atlassian.com/t5/Bitbucket-questions/remote-url-in-Bitbucket-server-what-does-scm-represent-is-it/qaq-p/2060987
 	id = strings.TrimPrefix(id, "scm/")
+
+	// Handle context scenarios --> https://confluence.atlassian.com/bitbucketserver/change-bitbucket-s-context-path-776640153.html
+	idsplit := strings.Split(id, "/scm/")
```

Comment:
Quick and easy fix, trying to trim `scm/` and then trying to split on `/scm/` in case the trim did not work. However for me it was a bit hard to understand why this worked at all, maybe something like

```
	// For servers without a http context path, we simply trim the 'scm' part
	if strings.HasPrefix(id, "scm/") {
		id = strings.TrimPrefix(id, "scm/")
		// For servers that use 'scm' as the http context path (hopefully no one has that), we can not do anything
		if strings.Contains(id, "scm/") {
			return "", "", fmt.Errorf("Context path 'scm/' in git address is reserved and not allowed: %s", s)
		}
		// For servers that use a http context path, we determine the context by spliting at '/scm/' and using the left part as the path
	} else if strings.Contains(id, "/scm/") {
		idsplit := strings.Split(id, "/scm/")
		host = host + "/" + idsplit[0]
		id = idsplit[1]
		// No other servers are supported, SSH URLs only work when we make the assumption that the SSH server runs at port 22 and the http server with https at port 443, which is not always the case
	} else {
		return "", "", fmt.Errorf("could not parse git address, note that SSH URLs are not supported: %s", s)
	}
```

would be easier to read?

Granted, it is more code, but I think it also gives a clearer feedback to the user when an unsupported value has been passed.

I've made a small example here: https://go.dev/play/p/QQLFmGZcbzK

Reply:
I have now added code to cover all cases possible. The new code will cover:

- muliple levels of context paths
- /scm/ in context path
- ssh path supplied will result in error instead of panic